specification/07-theory_operations.adoc

```diff
@@ -525,15 +525,62 @@ host supervisor domain manager may directly assign a TDI to a TVM, through the
-2. The TVM verifies and accepts the locked TDI into its TCB.
-3. The TVM asks the TSM to move the TDI to the TDISP `RUN` state.
+   having the TSM move the TDI into the TDISP `CONFIG_LOCKED` state. This is
+   achieved through the `sbi_covh_add_interface()` `COVH` ABI.
+2. The TVM xref:_tdi_acceptation[verifies and accepts the locked TDI] into its
+   TCB.
+3. The TVM asks the TSM to move the TDI to the TDISP `RUN` state, by calling the
+   `sbi_covg_run_interface()` `COVG` ABI.
 4. The TVM verifies that the TDI is in the TDISP `RUN` state and starts
-   using it.
-
-The next section describe in more details what step 2 covers, i.e. which steps
-a TVM should follow in order to be able to verify a locked TDI and accept or
-reject it into its TCB.
+   using it. This verification is provided by the
+   `sbi_covg_get_interface_state()` `COVG` ABI.
+
+The next two sections respectively give a detailed description of the overall
+assignment process and one if its most critical steps: the TVM decision of
+accepting or rejectin the TDI into its TCB.
+
+==== Assignment Flow
+
+Assigning an interface to a TVM goes through the following steps:
+
+1. The host supervisor domain manager calls the `sbi_covh_add_interface()`
+   `COVH` function to initiate the TDI assignment. The TSM rejects that request
+   if a secured SPDM session is not established with the DSM or if a stream IDE
+   link is not set up.
+2. The TSM generates and sends a TDISP `LOCK_INTERFACE_REQUEST` message to the
+   DSM, which is encapsulated into a vendor-defined SPDM request.
+3. The TSM receives the TDISP `LOCK_INTERFACE_RESPONSE` message, which contains
+   a device-generated nonce. The TSM stores the nonce.
+4. The TSM programs the DMA mappings into the trusted IOMMU space, in order for
+   the TDI inbound transactions to map into the TVM address space. The IOMMU
+   mappings are programmed but not validated. They will be validated only once
+   the TDI moves to the TDISP `RUN` state, i.e. when it accepts the TDI into
+   its TCB through the `sbi_covg_run_interface()` `COVG` ABI.
+5. For each of the assigned TDI MMIO range, the host supervisor domain manager
+   calls the `sbi_covh_add_tvm_device_region()` `COVH` ABI. The TSM creates
+   uncacheable G-stage mappings for each range so that the TVM outbound
+   transaction will not trigger MMIO page faults. Here as well, the TSM does
+   not enable the above mappings until the TVM moves the TDI to the TDISP `RUN`
+   state.
+6. Through regular discovery mechanisms (ACPI, PCI bus scanning), the TVM
+   detects the assigned TDI. It is important to note that the TDI configuration
+   space, including its BARs, is emulated by the host security domain manager.
+   The TDI MMIO ranges are mapped into the TVM address space by the the host
+   security domain manager itself.
+7. Before using the TDI, the TVM must
+   xref:tdi-acceptation[accept it into its TCB]. Moreover, the TVM must not use
```

Comment:
The current tdi-acceptation section does not have any flow. Such as when GET_MEASUREMENT is sent, when GET_INTERFACE_REPORT is sent.  

Do we need to address that?

Reply:
I added a sequence diagram for the verification and the acceptation flows. Let me know how it looks.